accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+This can be done by supplying the command to the Management Node via Salt.
+
+### CVE Auditing
+
+By leveraging the existing feature of CVE Auditing, SUSE Manager can detect and notify the user whether the cluster is running a vulnerable image. When SUSE Manager rebuilds the image and patches it, SUSE Manager can also guide the user in modifying the Kubernetes deployment to pull the updated image (in case `imagePullPolicy:` is not `Always`) or do a `kubectl rolling-update`.
+
+## Future work
+
+### Deploy OpenStack Cloud
+
+The approach described in this RFC can be further reused for deploying and managing an OpenStack Cloud using AirShip, given that `skuba` is similar to `airshipctl`:
+
+- provision nodes to be part of the cloud
+- tag nodes and bootstrap them to be part of the cloud
+- manage the lifecycle of the cloud
+
+### Run other products in the cluster
+
+It would also be possible, after that the cluster has been deployed, to deploy other SUSE products on top of the cluster (e.g. SUSE OpenStack Cloud, SUSE Enterprise storage): every product would have a deployment descriptor on top of CaaS Platform, and SUSE Manager would only have to deploy CaaS Platform and then the deployment of the product.
+
+In this case, SUSE Manager would be the central point of deployment for other products based on CaaS Platform and show the health of every cluster installed, while offering the possibility of managing and upgrading the cluster.
+
+### Kubernetes dashboard
+
+In the future developments, SUSE Manager can also embed the Kubernetes dashboard to help the user in deploy applications on top of CaaS Platform.
+The alternative of offering to configure application deployments from SUSE Manager would be seen as a second-choice solution compared to Kubernetes dashboard.
+
+### Re-use of deployment structure
+
+A small part of the deployment of the CaaS Platform can be reused for other Kubernetes distributions (e.g. OpenShift); deploying on bare metal requires machines configured with very specific requirements and that can be solved with AutoYaST (or Kickstart in the Red Hat case).
+
+Every distribution uses then its deployment helper (OpenShift uses `openshift-install`) to bootstrap the cluster. In this regard, `skuba` is only targeting SUSE CaaS Platform and cannot be reused for bootstrapping and managing a cluster.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+What if the customer wants to deploy another Kubernetes distribution?
+There will not be complete support for its deployment, or at least there is only support for the deployment on bare metal.
+Also, there is not support for managing this cluster because we are relying on `skuba`.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Managing the cluster with `kubectl` and importing its `kubeconfig` without using `skuba`: limited functionality, every Kubernetes distribution must be treated according to its specification, lots of duplicate work.
```

Comment:
To a big extent, this is where we are trying to head with `cluster-api` and the SUSE Application Lifecycle operator that could handle different products on top of Kubernetes.

Reply:
Does `cluster-api` still require `skuba`? Would it be able to handle non-CaaS Platform clusters?

Reply:
It won't require skuba and it will be able to handle non-CaaSP clusters (as long as they provide an actuator for the cluster-api)